### ISSUE TITLE
Classification performance metrics

### ISSUE THREAD

**Issue by MUST-Deep-Learning** [MUST-Deep-Learning/KnowIt#189]
For classification tasks, in `KnowIt.trainer.model_config.PLModel._compute_performance` where the model predictions and ground truths are passed to the performance metric function from torchmetrics it is possible that some of the metrics expect indices as predictions and targets and KnowIt provides OHE'd values by default.

It is worth checking that this does not happen and that the required conversion occurs if needed.

Additionally: In the case that it is a binary classification task, the order of the two classes needs to be controlled for example if calculating precision or recall. We do not currently enforce any such ordering.

Additionally: Check that these issues do not also occur for some loss functions (cross entropy).

**Comment by tiantheunissen** [comment]
Note that in the latest pull request (to be merged soon) I incorporated the OHE of target values for "accuracy" and included a method "requires_ohe" in the fetch_torch_mods.py script for adding other performance metrics that might require the OHE of targets.

I have not tested this with the binary classifier yet.
